assert/assertion_order.go

```diff
@@ -9,7 +9,7 @@ import (
 func isOrdered(t TestingT, object interface{}, allowedComparesResults []CompareType, failMessage string, msgAndArgs ...interface{}) bool {
```

Comment:
`isOrdered` is now an `Helper`. It must have the helper block before calling Fail.

Reply:
The call to Helper must be inside `IsDecreasing()` et al, not inside `isOrdered()`, this is not a regression as the functions are not currently helpers. We should have a separate issue to add `Helper()` calls to exported functions which need them.

Reply:
> The call to Helper must be inside IsDecreasing() et al, not inside isOrdered()

`isOrdered` now calls `assert.Fail` so errors will be reported there. But that not what we want. It is now an `Helper` while it wasn't the case before.

> this is not a regression as the functions are not currently helpers.

But you are changing the behaviour.

> We should have a separate issue to add Helper() calls to exported functions which need them.

This is assuming that the Testify project had ressources to do such a task. In the current state I'm the only active maintainer and I'm not allowed to merge my own code. So I can't wait for a community effort to do it.

So I'm asking again to fix this.